src/systems/triggered_publisher/TriggeredPublisher.hh

```diff
@@ -210,21 +228,49 @@ namespace systems
       transport::Node::Publisher pub;
     };
 
+    /// \brief Class that holds necessary bits for each specified service output
+    private: struct ServOutputInfo
+    {
+      /// \brief Service name
+      std::string srvName;
+
+      /// \brief Service request type
+      std::string reqType;
+
+      /// \brief Service reply type
+      std::string repType;
+
+      /// \brief Service request message
+      std::string reqMsg;
+
+      /// \brief Serivce timeout
+      int timeout;
+    };
+
     /// \brief List of InputMatchers
     private: std::vector<std::unique_ptr<InputMatcher>> matchers;
 
     /// \brief List of outputs
     private: std::vector<OutputInfo> outputInfo;
 
+    /// \brief List of service outputs
+    private: std::vector<ServOutputInfo> servOutputInfo;
+
     /// \brief Ignition communication node.
     private: transport::Node node;
 
     /// \brief Counter that tells the publisher how many times to publish
     private: std::size_t publishCount{0};
 
+    /// \brief Flag to trigger calling a service in the DoWork thread
+    private: bool triggerServ{false};
+
     /// \brief Mutex to synchronize access to publishCount
     private: std::mutex publishCountMutex;
 
+    /// \brief Mutex to synchronize access to serviceCount
+    private: std::mutex triggerServMutex;
```

Comment:
Nit: I know this was already missing, but I would add `#include <mutex>` at the top of the file here.